modules/tinymce/src/core/main/ts/content/GetContentImpl.ts

```diff
@@ -23,38 +23,36 @@ const trimEmptyContents = (editor: Editor, html: string): string => {
 };
 
 const getContentFromBody = (editor: Editor, args: GetContentArgs, format: ContentFormat, body: HTMLElement): Content => {
-  let content;
-
-  args.format = format;
-  args.get = true;
-  args.getInner = true;
-
-  if (!args.no_events) {
-    editor.fire('BeforeGetContent', args);
-  }
-
-  if (args.format === 'raw') {
+  const updatedArgs = args.no_events ? args : editor.fire('BeforeGetContent', {
```

Comment:
Yeah, this mutation is something I wanted to get rid of for a long time but I've been afraid that implementations out there would break. Is this something we could schedule for 6.0 cloning these instead of letting the event handlers mutate things.

Reply:
Yeah, I had the same concern. When we discussed this at standup though with Product, QA, and the rest of the editor team, we decided the implications of it mutating were worse than things potentially breaking for something that's not documented, nor really expected. Additionally, it's not that common for people to be firing custom events, they'd generally normally listen instead for existing events. It's also a relatively simple fix if someone was incorrectly relying on the mutation :shrug: 

I should add from that meeting, we did mention this would definitely have to be called out in the release notes with the workaround for anyone relying on it. That said, I'll bring it up again at standup today and make 100% sure we are still happy to go along with this.

Reply:
I'm happy to go with a fix in 5.9 given the the fix for anyone who is relying on Mutation is small and the benefits it provides.  We will do our best to alert to the community of the potential breaking changes if they rely on event mutation, we will pin a message on github issues as an early warning followed up by Docs.